moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -849,7 +848,6 @@ Eigen::VectorXd Trajectory::getAcceleration(double time) const
   const double acceleration =
       2.0 * (it->path_pos_ - previous->path_pos_ - time_step * previous->path_vel_) / (time_step * time_step);
 
-  time_step = time - previous->time_;
```

Comment:
Notice that a different `time_step` is used previously at L848. The `time_step` at L848 is exactly at one of the waypoints that TOTG used in calculation so vel/accel limits aren't exceeded.

The `time_step` at L852 is not exactly on one of the waypoints that TOTG used in calculation and that's why it exceeds the vel/accel limits.

There will be a very small loss in accuracy by removing this but it should be fine if TOTG is run with a small `delta_t`. Since the default `delta_t` is 0.001sec, that's certainly true.

Reply:
Thanks for the explanation!